### PR TITLE
Фикс регистрации

### PR DESCRIPTION
**Pull request by ZefRi19**
https://trello.com/c/A5rW9BoP/21-%D0%B8%D1%81%D0%BF%D1%80%D0%B0%D0%B2%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F-%D0%BF%D0%BE-%D1%80%D0%B5%D0%B3%D0%B8%D1%81%D1%82%D1%80%D0%B0%D1%86%D0%B8%D0%B8

### CODE REVIEW

File: accounts/views.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 from django.http import HttpResponseRedirect
 from django.shortcuts import render
 from django.contrib.auth import authenticate, login
 
 
+def check_authenticated(user):
+    return not user.is_authenticated
+
+
+@method_decorator(user_passes_test(check_authenticated, login_url='/', redirect_field_name=''), name='dispatch')
```
Comment (lifearoundfreaks) [inline accounts/views.py]: 79 characters limit. Also, seems like too much code to check a single condition,

Comment (ZefRi19) [reply accounts/views.py]: dobe

File: accounts/views.py [hunk 2]
```diff
@@ -6,12 +7,17 @@
 from django.contrib.auth import authenticate, login
 
 
-class RegistrationView(FormView):
+class RegistrationView(AccessMixin, FormView):
```
Comment (lifearoundfreaks) [inline accounts/views.py:23]: Seems much better, Also use it on login view. And use LoginRequiredMixin on logout view.

Comment (ZefRi19) [reply accounts/views.py:23]: done

**Pull request merged**